tips/TIP-0018/tip-0018.md

```diff
@@ -0,0 +1,3755 @@
+function as global addresses for the state machines, but unlike <i>Ed25519 Addresses</i>, they are not backed by private
+keys that could be used for signing. The rightful owners who can unlock these addresses are defined in the outputs
+themselves.
+
+Since such addresses are accounts in the ledger, it is possible to send funds to these addresses. The unlock mechanism
+of such funds is designed in a way that **proving ownership of the address is reduced to the ability to unlock the
+corresponding output that defines the address.**
+
+### Alias Locking & Unlocking
+
+A transaction may consume a (non-alias) output that belongs to an <i>Alias Address</i> by also consuming (and thus
+unlocking) the alias output with the matching `Alias ID`. This serves the exact same purpose as providing a signature
+to unlock an output locked under a private key backed address, such as <i>Ed25519 Addresses</i>.
+
+On protocol level, alias unlocking is done using a new unlock block type, called **Alias Unlock Block**.
+
+<details>
+<summary>Alias Unlock Block</summary>
+<blockquote>
+        Points to the unlock block of a consumed alias output.
+</blockquote>
+</details>
+
+<table>
+    <tr>
+        <td><b>Name</b></td>
+        <td><b>Type</b></td>
+        <td><b>Description</b></td>
+    </tr>
+    <tr>
+        <td>Unlock Block Type</td>
+        <td>uint8</td>
+        <td>
+            Set to <strong>value 2</strong> to denote a <i>Alias Unlock Block</i>.
+        </td>
+    </tr>
+    <tr>
+        <td>Alias Reference Unlock Index</td>
+        <td>uint16</td>
+        <td>
+            Index of input and unlock block corresponding to an alias output.
+        </td>
+    </tr>
+</table>
+
+This unlock block is similar to the <i>Reference Unlock Block</i>. However, it is valid if and only if the input of the
+transaction at index `Alias Reference Unlock Index` is an alias output with the same `Alias ID` as the `Address` field
+of the to-be unlocked output.
+
+Additionally, the <i>Alias Unlock Blocks</i> must also be ordered to prevent circular dependencies:
+
+If the i-th *Unlock Block* of a transaction is an *Alias Unlock Block* and has `Alias Reference Unlock Index` set to k,
+it must hold that i > k. Hence, an <i>Alias Unlock Block</i> can only reference an *Unlock Block* (unlocking the
+corresponding alias) at a smaller index.
+
+For example the scenario where `Alias A` is locked to the address of `Alias B` while `Alias B` is in locked to the
+address of `Alias A` introduces a circular dependency and is not well-defined. By requiring the *Unlock Blocks* to be
+ordered as described above, a transaction consuming `Alias A` as well as `Alias B` can never be valid as there would
+always need to be one *Alias Unlock Block* referencing a greater index.
+
+#### Alias Unlock Block Syntactic Validation
+
+ - It must hold that 0 ≤ `Alias Reference Unlock Index` ≤ `Max Inputs Count`.
```

Comment:
Shouldn't this be `< Max Inputs Count` ?

Reply:
you are right!